core/src/saros/communication/connection/ConnectionHandler.java

```diff
@@ -177,6 +180,73 @@ public void setCallback(IConnectingFailureCallback callback) {
     this.callback = callback;
   }
 
+  // TODO make private, keep XMPPException in core, move messages to CoreMessages
+  public static String generateHumanReadableErrorMessage(
+      XMPPAccount account, XMPPException xmppException) {
+
+    // as of Smack 3.3.1 this is always null for connection attempts
+    // Throwable cause = e.getWrappedThrowable();
```

Comment:
Why do we keep this code?

Reply:
It is removed in a following commit. (I think the one you looked at was the one where I just copied the IntelliJ Method).